internal/project/project.go

```diff
@@ -929,7 +935,7 @@ func (p *Project) setRootFiles(rootFileNames []string) {
 		// !!! updateNonInferredProjectFiles uses a fileExists check, which I guess
 		// could be needed if a watcher fails?
 		newRootScriptInfos[path] = struct{}{}
-		p.rootFileNames.Set(path, file)
+		p.setRootFileName(path, file)
```

Comment:
this is used only by configured project instead of changing this may be rename the methods so its clear which project kind they are operating on